File: 2084-198-2466-count-ways-to-build-good-strings/2084-198-2466-count-ways-to-build-good-strings.cpp
class Solution {
public:
    int dp[100001];
    const int mod = 1e9+7;
    
    int countGoodStrings(int low, int high, int zero, int one) {
        for(int i = low; i<= high;i++)dp[i]= 1;
        for(int i = high ; i >= 0 ;i--){
            if(i+zero < 100001) dp[i]+=(dp[i+zero]);
            if(i+one < 100001) dp[i]+=(dp[i+one]);
            dp[i]%=mod;
        }
        return dp[0];
    }
};

File: 2084-198-2466-count-ways-to-build-good-strings/2084-198-2466-count-ways-to-build-good-strings.py
class Solution:
    def countGoodStrings(self, low: int, high: int, zero: int, one: int) -> int:
        mod = int(1e9 + 7)
        dp = [0] * (high + 1)
        
        for i in range(low, high + 1):
            dp[i] = 1

        for i in range(high, -1, -1):
            if i + zero <= high:
                dp[i] += dp[i + zero]
            if i + one <= high:
                dp[i] += dp[i + one]
            dp[i] %= mod
        
        return dp[0]
